source/FreeRTOS_IPv6_Utils.c

```diff
@@ -285,6 +285,141 @@ size_t usGetExtensionHeaderLength( const uint8_t * pucEthernetBuffer,
+ * This is a "convenience" function that keeps all these tasks under one roof for easier maintenance.
+ *
+ * @param[in] pxEndPoint The end-point for which a network up/down event is being handled.
+ * @param[in] xNetworkGoingUp pdTRUE when the network goes UP, pdFALSE when the network goes DOWN.
+ */
+void vManageSolicitedNodeAddress( struct xNetworkEndPoint * pxEndPoint,
+                                  BaseType_t xNetworkGoingUp )
+{
+    configASSERT( pxEndPoint != NULL );
+    configASSERT( pxEndPoint->pxNetworkInterface != NULL );
+
+    /* do{}while(0) to allow for the use of break statements */
+    do
+    {
+        IPv6_Type_t xAddressType = xIPv6_GetIPType( &( pxEndPoint->ipv6_settings.xIPAddress ) );
+        MACAddress_t xMACAddress = { {
+                                         0x33U,
+                                         0x33U,
+                                         0xFFU,
+                                         pxEndPoint->ipv6_settings.xIPAddress.ucBytes[ 13 ],
+                                         pxEndPoint->ipv6_settings.xIPAddress.ucBytes[ 14 ],
+                                         pxEndPoint->ipv6_settings.xIPAddress.ucBytes[ 15 ]
+                                     } };
+
+        /* During the very first network DOWN event, pxEndPoint->ipv6_settings does not yet hold the proper address and
+         * therefore the MAC address above will be incorrect as well, however nothing bad will happen, because the address
+         * type check below will kick us out before the call to pfRemoveAllowedMAC(). Without the check below, the network
+         * driver end up being called once to register 33:33:FF:00:00:00 and that MAC never gets unregistered. */
+
+        /* Solicited-node multicast addresses only apply to normal unicast non-loopback addresses. */
+        if( ( xAddressType != eIPv6_LinkLocal ) && ( xAddressType != eIPv6_SiteLocal ) && ( xAddressType != eIPv6_Global ) )
+        {
+            /* The address of this end-point is something other than a normal unicast address... Maybe it's the
+             * loopback address or maybe this is an error scenario. In any case, there is no corresponding
+             * solicited-node multicast address that we need to manage. Do nothing.*/
+            break;
+        }
+
+        /* Update the network driver filter */
+        if( xNetworkGoingUp == pdTRUE )
+        {
+            if( pxEndPoint->pxNetworkInterface->pfAddAllowedMAC != NULL )
+            {
+                pxEndPoint->pxNetworkInterface->pfAddAllowedMAC( pxEndPoint->pxNetworkInterface, xMACAddress.ucBytes );
+            }
+        }
+        else
+        {
+            if( pxEndPoint->pxNetworkInterface->pfRemoveAllowedMAC != NULL )
+            {
+                pxEndPoint->pxNetworkInterface->pfRemoveAllowedMAC( pxEndPoint->pxNetworkInterface, xMACAddress.ucBytes );
+            }
+        }
+
+        #if ( ipconfigIS_ENABLED( ipconfigSUPPORT_IP_MULTICAST ) )
```

Comment:
Is this part of the code supposed to be added as part of PR #1019 ?

Reply:
Yes, you're correct. Code removed.